poet/poet.py

```diff
@@ -76,10 +85,106 @@ def walk(package):
     return sorted(discovered)
 
 
-def research_package(name, version=None):
+class PipSourceMetadataException(Exception):
+    pass
+
+@dataclass
+class PackageMetadata:
+    name: str
+    homepage: str
+    url: str
+    checksum: str
+    checksum_type: str
+
+def get_download_url_from_pip_source_file(module: str, pip_source_file: Path, output_dir=os.getenv("PIP_SOURCE_DIR")) -> str:
```

Comment:
huh.  my first impulse is to say "please move the `os.getenv()` invocation out of the method signature and up to the top scope", something like
```python
OUTPUT_DIR: Path = Path(os.getenv("PIP_SOURCE_DIR"))
...
def get_download_url_from_pip_source_file(
    module: str,
    pip_source_file: Path,
    output_dir: Path = OUTPUT_DIR
) -> str:
```
since my notion is that `os.getenv()` would get called every time `get_download_url_from_pip_source_file()` is called, and thus the behavior of the program could change mid-run if the value of that env var changes.

but i don't know whether that's actually the behavior?

nevertheless i still like moving that `os.getenv()` to the top scope for aesthetic reasons :)

Reply:
Addressed in: [b8abb61](https://github.com/indigo-ag/homebrew-pypi-poet/pull/1/commits/b8abb6137f1b6d66ddc05e52473cc3c83aec98a9)